### PR TITLE
Fix sizeof call in memset

### PR DESCRIPTION
**Pull request by maryla-uc**

### CODE REVIEW

**Review by y-guyon — APPROVED** [review]

File: src/gainmap.c [hunk 1]
```diff
@@ -55,7 +55,7 @@ avifBool avifGainMapMetadataFractionsToDouble(avifGainMapMetadataDouble * dst, c
 
 static void avifGainMapMetadataSetDefaults(avifGainMapMetadataDouble * metadata)
 {
-    memset(metadata, 0, sizeof(avifGainMapMetadata));
+    memset(metadata, 0, sizeof(*metadata));
```
Comment (wantehchang) [inline src/gainmap.c:58]: The function name can begin with "avifGainMapMetadataDouble" to follow the naming convention,.

**Review by wantehchang — APPROVED** [review]
LGTM. We can look at our Coverity defects to see if it detected this bug.

**Review by wantehchang — APPROVED** [review]

**Comment by wantehchang** [comment]
I checked our Coverity defects. It did not seem to detect this error. Strange.

**Comment by maryla-uc** [comment]
I assume the bug was not caught because the two structs happen to have the exact same size.

**Pull request merged**